spec.md

```diff
@@ -2709,6 +2713,67 @@ need only respect the 2-state behavior of a circuit.  This is a limitation on
 the scope of what behavior is observable (i.e., a relaxation of the
 ["as-if"](https://en.wikipedia.org/wiki/As-if_rule) rule).
 
+## Indeterminate Values
+
+An indeterminate value represents a value which is unknown or unspecified.  
+Indeterminate values are generally implementation defined, with constraints 
+specified below.  An indeterminate value may be assumed to be any specific 
+value, at an implementation's discresion, if, in doing so, all observable 
+behavior is as if the indeterminate value always took the specific value.
```

Comment:
Just to clarify, you are stating that a "value" may not necessarily be a literal, yes?

It may be useful to change the example below to clearly indicate this. I assuming that we want to also allow optimization of:

```
wire a: UInt<8>
a is invalid
when c :
  a <= b
```

To:
```
a <= b
```

It would probably be better to change the example below to not use `h4` and instead use a non-literal value.

If we are choosing to NOT allow this, then the SFC (and MFC) are in violation of the spec.